### ISSUE TITLE
Rest API failing to handle optional URL parameters

### ISSUE THREAD

**Issue by ian-zunderdorp** [hasura/graphql-engine#8574]
### Version Information

Server Version: Tried 2.6.2, 2.7.0 and 2.8.0-beta.1
CLI Version (for CLI related issue): N/A

### Environment

OSS

### What is the expected behaviour?

For a GraphQL query where a parameter is optional, I would like to be able to configure a REST API connector to optionally utilize this field.
Dummy example: 
```graphql
type Query {
  getBlogPosts(type: [String]) : BlogPost
}
```

```graphql
query {
  allBlogPosts: getBlogPosts() {
    id
    title
  }

  tutorials: getBlogPosts(type: ["tutorial"]) {
    id
    title
  }
}
```

Connecting to a REST API at `example.org/blog`, I would like to be able to configure that allBlogPosts executes `GET example.org/blog` and tutorials `example.org/blog?type=tutorial`

### Keywords

REST connection optional nullable

### What is the current behaviour?

It currently cannot handle parameters being optional in any way: `allBlogPosts` in my example would give an error `Cannot interpolate type: 'Null'.`. It is not possible to handle that with the Kriti templating language either, since both if-statements and defaulting operator are not recognized as valid syntax.

### How to reproduce the issue?

1. Create a REST API that has optional parameters
2. In Hasura, add a new action
  a. Use an action definition where a parameter is optional
  b. Map that parameter to a query string parameter
3. Execute a query where the optional parameter is skipped

### Any possible solutions?

Hasura should be able to at least handle a Null value in order to create a "valid" HTTP request. Whether it is correctly interpreted by the receiving server is another issue.

In my example, the minimum behavior Hasura should be able to do is to construct `example.org/blog?type`.

Better behavior would be to be able to declare for every query parameter what the behavior should be on Null values: omit, include only as key (`?type`), include as key with no value (`?type=`).

### Can you identify the location in the source code where the problem exists?

I have not checked, we are currently looking into StepZen instead of Hasura since I have a working example in there.

### If the bug is confirmed, would you be willing to submit a PR?

As above, we're looking into alternative tooling, so it's highly unlikely that we're dedicating time to solve the bug.

**Comment by sordina** [comment]
Hi @ian-zunderdorp

I've confirmed that this behaviour is as you have described. We're looking into how to address this use case.

**Comment by solomon-b** [comment]
@ian-zunderdorp Would it solve your issue if you were able to use our defaulting operator inside string interpolations?

**Comment by ian-zunderdorp** [comment]
Probably it would. In the meantime we have already switched over to StepZen where we managed to do all mapping needed correctly.

**Comment by soorajshankar** [comment]
Related issue : https://github.com/hasura/graphql-engine/issues/7909

**Comment by paritosh-08** [comment]
Hey folks, this issue has been fixed and released as part of [v2.15.0-beta.1](https://github.com/hasura/graphql-engine/releases/tag/v2.15.0-beta.1).
